scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5082 @@
+        :math:`x = \log(1) = 0`. Because the log-CDF has range
+        :math:`[-\infty, 0]`, the inverse log-CDF is only defined on the
+        negative reals; for :math:`x > 0`, ``ilogcdf`` returns ``nan``.
+
+        Occasionally it is needed to find the argument of the CDF for which
+        the resulting probability is very close to `0` or `1` - too close to
+        represent accurately with floating point arithmetic. In many cases,
+        however, the *logarithm* of this resulting probability may be
+        represented in floating point arithmetic, in which case this function
+        may be used to find the argument of the CDF for which the *logarithm*
+        of the resulting probability is `x`.
+
+        Due to the inherent spacing between floating point numbers, it is common
+        for there to be no value of the argument for which the log-CDF is close
+        to the provided `x`. In such cases, this function returns NaN.
+
+        Examples
+        --------
+        Instantiate a distribution with the desired parameters:
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> X = stats.Uniform(a=-0.5, b=0.5)
+
+        Evaluate the inverse log-CDF at the desired argument:
+
+        >>> X.ilogcdf(-0.25)
+        0.2788007830714034
+        >>> np.allclose(X.ilogcdf(-0.25), X.icdf(np.exp(-0.25)))
+        True
+
+        This function may return NaN when the spacing between floating point
+        values is too coarse to represent an argument for which the log-CDF is
+        close to the provided `x`, even when the function is theoretically
+        continuous within the domain of the argument.
+
+        >>> X.ilogcdf(-np.inf), X.ilogcdf(-1e10), X.ilogcdf(-30.)
+        (-0.5, nan, -0.4999999999998863)
+
+        """
+        return self._ilogcdf_dispatch(x, method=method, **self._parameters)
+
+    @_dispatch
+    def _ilogcdf_dispatch(self, x, method=None, **kwargs):
+        if self._overrides('_ilogcdf_formula'):
+            method = self._ilogcdf_formula
+        elif self._overrides('_ilogccdf_formula'):
+            method = self._ilogcdf_complementarity
+        else:
+            method = self._ilogcdf_inversion
+        return method
+
+    def _ilogcdf_formula(self, x, **kwargs):
+        raise NotImplementedError(self._not_implemented)
+
+    def _ilogcdf_complementarity(self, x, **kwargs):
+        return self._ilogccdf_dispatch(_log1mexp(x), **kwargs)
+
+    def _ilogcdf_inversion(self, x, **kwargs):
+        return self._solve_bounded(self._logcdf_dispatch, x, kwargs=kwargs)
+
+    @_set_invalid_nan
+    def icdf(self, x, *, method=None):
+        r"""Inverse cumulative distribution function
+
+        The inverse cumulative distribution function :math:`F^{-1}(x)` is the
+        argument :math:`y` for which the cumulative distribution function
+        :math:`F(y)` evaluates to :math:`x`.
+
+        .. math::
+
+            F^{-1}(x) = y \text{ such that } F(y) = x`
```

Comment:
I thought about it, but I have gotten complaints about use of LaTeX before, so I thought this was a good compromise. That's why I've used leq/geq characters instead of `\leq`, etc.

Reply:
I'm all for well-written, readable TeX. Here it's wash, I'd use \text{\quad s.t. \quad} but remove quads on a complaint. Up to you really.

Reply:
Done.

Reply:
Need to fix; it should be  \quad \text{s.t.} \quad